### PR TITLE
Updates Product Scope, NFR and Glossary

### PR DESCRIPTION
**Pull request by rui-han-crh**
Updates Product Scope with Sectres relevant info
Updates NFR with some usability information
Updates Glossary to include definition of Notes and Secretaries

### CODE REVIEW

**Review by Neethesh26 — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -257,13 +257,15 @@ _{Explain here how the data archiving feature will be implemented}_
 
 **Target user profile**:
 
+* Secretaries and treasurers of clubs who need to keep track of a lot of paperwork and are looking for a convenient way to organise information.
```
Comment (rui-han-crh) [inline docs/DeveloperGuide.md:260]: Format inconsistency with below points, may need to rephrase this sentence

**Review by Pinran-J — APPROVED** [review]
LGTM

**Review by czhongwei — APPROVED** [review]
LGTM my g

**Pull request merged**